tests/python/helpers.py

```diff
@@ -1,45 +1,197 @@
+        # according to keypair index in account list
+        horizon = horizons[i % len(horizons)]
+        builder = Builder(env.name, horizon, MIN_FEE, kp.secret_seed)
+        builder.sequence = sequences[i]
+        builders.append(builder)
+
+    for b in builders:
+        logging.debug('created builder %s', b.address)
+
+    return builders
+
+
+def init_tx_builders(env, kps, sequences):
+    """Initialize transaction builders for each given seed."""
+    builders = []
+    for i, kp in enumerate(kps):
+        client = kin.KinClient(env)
+        builder = Builder(env.name, client.horizon, MIN_FEE, kp.secret_seed)
+        builder.sequence = sequences[i]
+        builders.append(builder)
+    return builders
+
+
+async def get(session: aiohttp.ClientSession, url, expected_status=200):
+    """Send an HTTP GET request and return response JSON data.
+
+    Fail if response isn't expected status code or format other than JSON.
+    """
+    async with session.get(url) as res:
+        res_data = await res.json()
+
+        if res.status != expected_status:
+            logging.error('Error in HTTP GET request to %s: %s', url, res_data)
+            raise RuntimeError('Error in HTTP GET request to {}'.format(url))
+
+    return res_data
 
-        for _ in await asyncio.gather(*promises):
-            return promises
+
+async def post(session: aiohttp.ClientSession, url, req_data, expected_statuses):
+    """Send an HTTP POST request with given data and return response JSON data.
+
+    Fail if response isn't expected status code or format other than JSON.
+
+    NOTE expected status is either OK 200 or Server Timeout 504 if the transaction
+    wasn't added to the next three ledgers.
+    """
+    async with session.post(url, data=req_data) as res:
+        res_data = await res.json()
+
+        if res.status not in expected_statuses:
+            logging.error('Error in HTTP POST request to %s with data %s: %s', url, req_data, res_data)
+            raise RuntimeError('Error in HTTP POST request to {}'.format(url))
+
+    return res_data
+
+
+class LoggingClientSession(aiohttp.ClientSession):
+    """aiohttp client session that logs requests."""
+
+    async def _request(self, method, url, **kwargs):
+        logging.debug('Starting request <%s %r>', method, url)
+        return await super()._request(method, url, **kwargs)
+
+
+async def send_txs(endpoint, xdrs, expected_statuses=[200, 504]):
+    """Send multiple async transaction XDRs to given endpoint."""
+    logging.info('sending %d transactions', len(xdrs))
+
+    async with LoggingClientSession(connector=aiohttp.TCPConnector(limit=5000)) as session:
+        url = '{}/transactions'.format(endpoint)
+        results = await asyncio.gather(*[post(session, url, {'tx': xdr.decode()}, expected_statuses) for xdr in xdrs])
+
+    logging.info('%d transactions sent', len(xdrs))
+    return results
+
+
+async def get_sequences(endpoint, addresses):
```

Comment:
I would consider changing it to ledger << 32 in the future when you have more time :smirk:

Reply:
okay :fire: